src/utils/ExtrusionJunction.h

```diff
@@ -0,0 +1,31 @@
+//Copyright (c) 2019 Ultimaker B.V.
+
+
+#ifndef UTILS_EXTRUSION_JUNCTION_H
+#define UTILS_EXTRUSION_JUNCTION_H
+
+#include "IntPoint.h"
+
+namespace arachne
+{
+
+struct ExtrusionJunction
```

Comment:
I'm not a fan of that. (Though I wouldn't necessarily have disapproved of it if it was that way in the first place.)

- This way, the variables are 'documented' in code by their names (or would be, if `w` had a better name).
- Tuples aren't as friendly w.r.t syntactic sugar as, say, Python. The way to get the second element of a tuple is `std::get<1>(tuple)`. There is no `operator[]`.
- I've seen @nallath move some functions that float around 'outside', that seemed to 'belong' to certain classes, since this code is a lot more C-style than we prefer. I think it would be wise to _at least_ wait with this change until _after_ such functions have all been moved/considered.